### PR TITLE
Support weekof

### PR DESCRIPTION
**Pull request by rubio41**
support week of date and month of date

### CODE REVIEW

File: Microsoft.Recognizers.Text.DateTime.Tests/English/TestDatePeriodExtractor.cs [hunk 1]
```diff
@@ -108,6 +107,11 @@ public void TestDatePeriodExtract()
             BasicTest("I'll leave summer", 11, 6);
             BasicTest("I'll leave summer 2016", 11, 11);
             BasicTest("I'll leave summer of 2016", 11, 14);
+
+            //test week of and month of
+            BasicTest("week of september.15th", 0, 22);
+            BasicTest("month of september.15th", 0, 23);
+
```
Comment (yiylin) [inline Microsoft.Recognizers.Text.DateTime.Tests/English/TestDatePeriodExtractor.cs]: Is "week of september.15th" the same as "week of september 15th"?

File: Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs [hunk 2]
```diff
@@ -19,41 +19,110 @@ public EnglishHolidayParserConfiguration() : base()
...
         private static DateObject WhiteLoverDay(int year) => new DateObject(year, 3, 14);
-        private static DateObject ChineseMilbuildday(int year) => new DateObject(year, 8, 1);
         private static DateObject FoolDay(int year) => new DateObject(year, 4, 1);
         private static DateObject GirlsDay(int year) => new DateObject(year, 3, 7);
         private static DateObject TreePlantDay(int year) => new DateObject(year, 3, 12);
         private static DateObject FemaleDay(int year) => new DateObject(year, 3, 8);
         private static DateObject ChildrenDay(int year) => new DateObject(year, 6, 1);
         private static DateObject YouthDay(int year) => new DateObject(year, 5, 4);
         private static DateObject TeacherDay(int year) => new DateObject(year, 9, 10);
-        private static DateObject AllSaintsDay(int year) => new DateObject(year, 10, 31);
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs]: Why was this removed?

File: Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs [hunk 3]
```diff
@@ -19,41 +19,110 @@ public EnglishHolidayParserConfiguration() : base()
...
+                {"allsoulsday", AllSoulsday},
+                {"guyfawkesday", GuyFawkesDay},
+                {"veteransday", Veteransday},
+                {"christmaseve", ChristmasEve},
+                {"newyeareve", NewYearEve},
             };
         }
         
-        private static DateObject Yuandan(int year) => new DateObject(year, 1, 1);
-        private static DateObject ChineseNationalDay(int year) => new DateObject(year, 10, 1);
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs]: Why were these removed too?

File: Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs (hunk 2 above)
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs]: BTW, what about holidays that change day like Easter or Chinese New Year?

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs [hunk 4]
```diff
@@ -125,6 +125,17 @@ public class SpanishDatePeriodExtractorConfiguration : IDatePeriodExtractorConfi
...
+        public static readonly Regex WeekOfRegex =
+            new Regex(
+                $@"(week)(\s*)(of)",
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+        public static readonly Regex MonthOfRegex =
+            new Regex(
+                $@"(month)(\s*)(of)",
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
```
Comment (yiylin) [inline Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs]: @tellarin Please remember to replace this.

File: Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs [hunk 5]
```diff
@@ -102,6 +103,40 @@ private List<Token> MergeTwoTimePoints(string text)
             return ret;
         }
 
+        private List<Token> SingleTimePointWithPatterns(string text)
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs]: Can you add a comment in the code giving an example of the cases that would be covered here?

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs (hunk 4 above)
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs]: There should be a TODO in the code here.

File: Microsoft.Recognizers.Text.DateTime.Tests/English/TestDatePeriodExtractor.cs (hunk 1 above)
Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime.Tests/English/TestDatePeriodExtractor.cs]: yes

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs (hunk 4 above)
Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs]: there are todo items around the code

File: Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs (hunk 3 above)
Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs]: rename yuandan to newyear. Didn't extract Chinese National Day in English holidays. It is supported in Chinese holidays.

File: Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs (hunk 2 above)
Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs]: All Saint Day=Halloween

Comment (rubio41) [inline Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs]: It cannot support the resolution of Easter day and Chinese New year now. In Chinese holiday parser, we will return Lunar flag for Chinese new year. But for Easter day, we don't have a solution yet.
